client/src/cordova/apple/xcode/ios/Outline/Classes/AppKitBridge/AppKitController.swift

```diff
@@ -16,13 +16,72 @@ import AppKit
 import ServiceManagement
 
 class AppKitController: NSObject {
+    static let shared = AppKitController()
     private var statusItemController: StatusItemController?
+    private var windowCloseObservers: [NSWindow: NSObjectProtocol] = [:]
 
     override public required init() {
         super.init()
-
-        // Indicates that the application is an ordinary app that appears in the Dock and may have a user interface.
         NSApp.setActivationPolicy(.regular)
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(windowDidBecomeMain(_:)),
+            name: NSWindow.didBecomeMainNotification,
+            object: nil
+        )
+        // After the app has fully launched, forcibly attach an observer to the initial UIWindow.
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1.0) {
+            for window in NSApp.windows {
+                if String(describing: window).contains("UINSWindow") {
+                    self.observeWindowClose(for: window)
```

Comment:
Why are there multiple calls to `observeWindowClose`? For example, L76 in this file and L100 in `StatusItemController.swift`. Can we reduce the calls?